### PR TITLE
Make tokens script multichain

### PR DESCRIPTION
**Pull request by antooni**
- [x] Write a script to ensure all tokens were correctly migrated 
- [x] Add docs for adding a chain and token

Resolves L2B-3503
Resolves L2B-3584

### CODE REVIEW

File: packages/config/src/tokens/script.ts [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  }
+
+  const sorted = sortByChainAndName(result)
+  logger.success('\nSorted ', 'tokens by chain and name')
+
+  saveResults(sorted)
+  logger.success('\nSaved ', 'output file')
+}
+
+function getSourceFile(logger: ScriptLogger) {
```
Comment (sz-piotr) [inline packages/config/src/tokens/script.ts:91]: readTokens

File: packages/config/src/tokens/script.ts [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  }) as Record<string, string>
+  logger.assert(errors.length === 0, 'Cannot parse source.jsonc')
+  const source = Source.parse(parsed)
+
+  logger.success('Loaded ', 'source file')
+
+  return source
+}
+
+function getOutputFile(logger: ScriptLogger) {
```
Comment (sz-piotr) [inline packages/config/src/tokens/script.ts:107]: readGenerated

File: packages/config/src/tokens/script.ts [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+function getCoingeckoClient() {
+  const env = getEnv()
+  const coingeckoApiKey = env.optionalString('COINGECKO_API_KEY')
+  const http = new HttpClient()
+  const coingeckoClient = new CoingeckoClient(http, coingeckoApiKey)
+  return coingeckoClient
+}
+
+function getChainConfiguration(logger: ScriptLogger, devId: string) {
+  const chain = chains.find((c) => c.devId === devId) // handle manta pacific case
```
Comment (sz-piotr) [inline packages/config/src/tokens/script.ts:127]: outdated comment

File: packages/config/src/tokens/script.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+      const type = getType(tokenLogger, devId, token)
+      const formula = getFormula(tokenLogger, devId, token)
+      const category = token.category ?? 'other'
+
+      const existingToken = findTokenInOutput(output, chainId, token)
+
+      if (existingToken) {
+        tokenLogger.skipping()
+        result.push({
```
Comment (sz-piotr) [inline packages/config/src/tokens/script.ts:43]: Add commend about config values overriding output

Comment (sz-piotr) [reply packages/config/src/tokens/script.ts:43]: Maybe coingeckoId here is missing

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/config/src/tokens/utils/ScriptLogger.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      ...options,
+    }
+  }
+
+  fetching(...messages: string[]) {
+    this.notify('Fetching...', ...messages)
+  }
+
+  processing(...messages: string[]) {
+    if (this.options.enabled) console.log('\n')
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/config/src/tokens/utils/ScriptLogger.ts]: b4edad40599fcc05f9ac68cdf104f30a0b1cc9bb

File: packages/config/src/tokens/utils/getCoingeckoId.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  logger: ScriptLogger,
+  address: EthereumAddress,
+  platform: string | undefined,
+) {
+  logger.assert(
+    platform !== undefined,
+    'Could not find coingecko platform identifier. Please add it chain config of the project',
+  )
+
+  const coinList = await coingeckoClient.getCoinList({
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/config/src/tokens/utils/getCoingeckoId.ts]: ee128985b5d79f4df6368bc819cbce0624f16f22

File: packages/config/src/tokens/types.ts [hunk 7]
```diff
@@ -1,49 +1,40 @@
...
-    }))
-}
+  bridgedUsing: z.optional(
+    z.object({
+      bridge: z.string(),
+      slug: z.string().optional(),
+    }),
+  ),
+})
+export type SourceEntry = z.infer<typeof SourceEntry>
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/config/src/tokens/types.ts]: 805a9561fc4ebee78876bad39d3c50c7c036d39f

File: packages/config/src/tokens/types.ts [hunk 8]
```diff
@@ -1,49 +1,40 @@
...
+export const Source = z.record(z.array(SourceEntry))
+export const Output = z.object({
+  comment: z.string().optional(),
+  tokens: z.array(Token),
+})
+export type Output = z.infer<typeof Output>
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/config/src/tokens/types.ts]: 805a9561fc4ebee78876bad39d3c50c7c036d39f

File: packages/config/package.json [hunk 9]
```diff
@@ -16,7 +16,8 @@
     "get-lz-data": "node -r esbuild-register scripts/omnichain",
     "update-multichain": "node -r esbuild-register scripts/multichain",
     "check-verified-contracts": "node -r esbuild-register scripts/checkVerifiedContracts",
-    "tokens:add": "node -r esbuild-register scripts/tokens/add"
+    "tokens": "node -r esbuild-register src/tokens/script.ts",
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/config/package.json]: fa421e1e4ba30b5f55a8bd6a7089805d1c5e0c82

File: packages/config/package.json [hunk 10]
```diff
@@ -16,7 +16,8 @@
...
+    "tokens:check-holes": "node -r esbuild-register scripts/detectCoingeckoPriceHoles.ts 4"
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/config/package.json]: a28f8b3e016c9aa01cdc9fe6f8a42e6028793564

File: packages/shared-pure/src/types/Token.ts [hunk 11]
```diff
@@ -1,31 +1,42 @@
...
+      bridge: z.string(),
+      slug: z.string().optional(),
+    }),
+  ),
+})
 
-  /** @deprecated */
-  category: 'ether' | 'stablecoin' | 'other'
-}
+export type Token = z.infer<typeof Token>
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/shared-pure/src/types/Token.ts]: 1f2e1e89c7b35b85f5f6e7f345cb36419934da1d

File: packages/frontend/src/utils/project/getProjectTvlTooltipText.ts [hunk 12]
```diff
@@ -1,14 +1,21 @@
-import { Layer2 } from '@l2beat/config'
-import { AssetType } from '@l2beat/shared-pure'
+import { Layer2, tokenList } from '@l2beat/config'
+import { AssetType, ChainId } from '@l2beat/shared-pure'
 
 import { languageJoin } from '../utils'
 
-export function getProjectTvlTooltipText(config: Layer2['config']) {
+export function getProjectTvlTooltipText(
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/frontend/src/utils/project/getProjectTvlTooltipText.ts]: 62705270151ed5796ad8b5e457925a8d9368c7dd

File: packages/frontend/src/pages/scaling/summary/props/getScalingSummaryView.ts [hunk 13]
```diff
@@ -74,7 +74,10 @@ function getScalingL2SummaryEntry(
             displayValue: formatUSD(stats.latestTvl),
           }
         : undefined,
-    tvlTooltip: getProjectTvlTooltipText(project.config),
+    tvlTooltip: getProjectTvlTooltipText(
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/frontend/src/pages/scaling/summary/props/getScalingSummaryView.ts]: 62705270151ed5796ad8b5e457925a8d9368c7dd

File: packages/config/src/tokens/script.ts (hunk 1 above)
Comment (antooni) [reply packages/config/src/tokens/script.ts]: 102aa6f1bdc831fc06e4789cf71cb206bcef302b

File: packages/config/src/tokens/script.ts (hunk 4 above)
Comment (antooni) [reply packages/config/src/tokens/script.ts]: c72dc72c12df62ee678aa14dcb0340efa71bb659

**Review by sz-piotr — APPROVED** [review]

**Pull request merged**